src/Common/src/CommandLine/CommandLineHelpers.cs

```diff
@@ -31,6 +31,11 @@ public static void AppendExpandedPaths(Dictionary<string, string> dictionary, st
                     string fullFileName = Path.GetFullPath(fileName);
 
                     string simpleName = Path.GetFileNameWithoutExtension(fileName);
+                    if (StringComparer.OrdinalIgnoreCase.Equals(Path.GetExtension(simpleName), ".ni"))
```

Comment:
This does not sounds like the right place for this logic.

Reply:
Hmm, this seems to be the place that currently translates OS paths into simple assembly names. Could you please elaborate on how you'd like me to change this logic?

Reply:
Why do we need ILC to know about .ni extension in the first place? AFAIK we don't ship .ni files anywhere, so we will not encounter them unless we put ourselves into such position. The `.ni` string in special cased in lots of places in this repo now. Do we really need them?

Reply:
Well, I think there are two different questions involved:

1) Should we try to reverse translate OS paths into assembly names?
2) If answer to (1) is positive, should the translation honor the .ni extension?

I actually don't know what is the right answer to the first question. The problem may well be that we're conflating file names used for uniqueness as keys in the _inputFilePaths dictionary with simple assembly names in GetModuleForSimpleName; perhaps GetModuleForSimpleName should be using a different lookup table based on logical assembly names found in the MSIL info.

For (2), JanV raised a similar question some time ago. I guess that, for testing purposes, it's kind of handy that the .ni extension ensures that, even within the same folder, Crossgen/CPAOT R2R compilation reads an "input file" (the MSIL dll) and outputs a different "output file" (the R2R dll). Recalling past trouble with FxTransforms in .NET Native, it always sucks when a build process rewrites the input with the output; but perhaps it doesn't suck so much if R2R compilation is idempotent and it should be (I think CPAOT isn't right now but that's something we should fix).

Reply:
I agree we should remove the .ni. extension and just have it not be a thing. I made basically this change for the `dotnet publish` work. See https://github.com/nattress/corert/commit/a0e9009772596fc6266d904f3479be28a8782e5c

Reply:
If other principal CoreRT architects like @jkotas or @MichalStrehovsky don't object, I'm not opposed to eradicating the .ni extension. I currently don't have enough expertise to assess all implications of such a move in advance though.

Reply:
Crossgen does not probe for .ni.dll files. If you place Foo.ni.dll into `/Platform_Assemblies_Paths`, it will not look at it when looking for assembly Foo. Simon's fix looks like a good solution to me.

The code proposed here would break user code if user decides to name their assembly Foo.ni.

Reply:
I don't want to hold up your check-in so we can coordinate our changes. You check it in like this if it allows us to get all the framework compiling and I'll pull it back out with my remove all `.ni` change. Sound good?

Reply:
Hmm, I'm afraid that my field access change will blow up without Michal's counterpart change to fix the offsets. I'm actually going through another round of testing as I moved a combo of this and Michal's change to another CoreRT checkout and I'm hitting some weird errors I'd like to understand first. Maybe it would be easier if you first merged the removal of "ni" as that's a trivial change; in the meantime, I'll get my head around the testing and I'll publish an update removing the .ni change that Jan and Michal didn't like.

Reply:
I have rolled back the ".ni" aspect of this change in the 4th commit.